test/functional/tests/fault_injection/test_fault_injection_with_mounted_core.py

```diff
@@ -166,3 +166,39 @@ def test_add_occupied_core(cache_mode):
 
     with TestRun.step("Stop all caches."):
         casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_add_twice_same_core(cache_mode):
+    """
+        title: Fault injection test if cache not accepts twice the same device as core.
+        description: |
+          Negative test of the ability to add the same core twice to cache.
+        pass_criteria:
+          - No system crash.
+          - The same core device cannot be used twice in one cache.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start OpenCAS"):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+
+    with TestRun.step("Add core device to OpenCAS."):
+        cache.add_core(core_dev)
+
+    with TestRun.step("Try add second time the same core device to OpenCAS"):
+        output = TestRun.executor.run_expect_fail(cli.add_core_cmd(cache_id=str(cache.cache_id),
+                                                                   core_dev=str(core_dev)))
+        if not output.stderr:
```

Comment:
How about checking if expected err msg is printed as well to be sure that it failed because of expected reason?